internal/dag/builder.go

```diff
@@ -527,9 +528,16 @@ func (b *Builder) computeRoutes(sw *ObjectStatusWriter, proxy *projcontour.HTTPP
 
 	visited = append(visited, proxy)
 	var routes []*Route
-	// Loop over and process all includes
-	for _, include := range proxy.Spec.Includes {
 
+	// Check for duplicate conditions on the includes
+	includes := proxy.Spec.Includes
+	if b.includeConditionsIdentical(includes) {
```

Comment:
What happens if the same prefix condition is used on a route and an include?

Reply:
What if we add `func (p *HttpProxySpec) AllConditions() []Condition` to get all the include and route conditions for checking?

Reply:
Defining duplicate routes in a Spec.Match is a possibility that we should look into (I'll open an issue). My biggest concern with this PR was to solve the multi-team component where folks could define the same include across different teams (which is one of our big selling points).

Reply:
https://github.com/projectcontour/contour/issues/1834